tests/functional/materializations/materialized_view_tests/test_materialized_view.py

```diff
@@ -0,0 +1,143 @@
+
+        # poll database
+        table_mid = get_row_count(project, "base_table")
+        view_mid = get_row_count(project, "base_materialized_view")
+
+        # refresh the materialized view
+        run_model("base_materialized_view")
+
+        # poll database
+        table_end = get_row_count(project, "base_table")
+        view_end = get_row_count(project, "base_materialized_view")
+
+        # new records were inserted in the table but didn't show up in the view until it was refreshed
+        assert table_start < table_mid == table_end
+        assert view_start == view_mid < view_end
+
+
+class OnConfigurationChangeCommon(PostgresOnConfigurationChangeBase):
+    def test_full_refresh_takes_precedence_over_any_configuration_changes(
+        self, configuration_changes, replace_message, configuration_change_message
+    ):
+        results, logs = run_model("base_materialized_view", full_refresh=True)
+        assert_proper_scenario(
+            self.on_configuration_change,
+            results,
+            logs,
+            RunStatus.Success,
+            messages_in_logs=[replace_message],
+            messages_not_in_logs=[configuration_change_message],
+        )
+
+    def test_model_is_refreshed_with_no_configuration_changes(
+        self, refresh_message, configuration_change_message
+    ):
+        results, logs = run_model("base_materialized_view")
+        assert_proper_scenario(
+            self.on_configuration_change,
+            results,
+            logs,
+            RunStatus.Success,
+            messages_in_logs=[refresh_message, configuration_change_message],
+        )
+
+
+class TestOnConfigurationChangeApply(OnConfigurationChangeCommon):
+    def test_model_applies_changes_with_configuration_changes(
+        self, configuration_changes, alter_message, update_index_message
+    ):
+        results, logs = run_model("base_materialized_view")
+        assert_proper_scenario(
+            self.on_configuration_change,
+            results,
+            logs,
+            RunStatus.Success,
+            messages_in_logs=[alter_message, update_index_message],
+        )
+
+
+class TestOnConfigurationChangeContinue(OnConfigurationChangeCommon):
+
+    on_configuration_change = OnConfigurationChangeOption.Continue
+
+    def test_model_is_not_refreshed_with_configuration_changes(
+        self, configuration_changes, configuration_change_continue_message, refresh_message
+    ):
+        results, logs = run_model("base_materialized_view")
+        assert_proper_scenario(
+            self.on_configuration_change,
+            results,
+            logs,
+            RunStatus.Success,
+            messages_in_logs=[configuration_change_continue_message],
+            messages_not_in_logs=[refresh_message],
+        )
+
+
+class TestOnConfigurationChangeFail(OnConfigurationChangeCommon):
+
+    on_configuration_change = OnConfigurationChangeOption.Fail
```

Comment:
nit: inlining `on_configuration_change` in the `assert_proper_scenario` call below here + in other tests that inherit from `OnConfigurationChangeCommon` (similar to RunStatus values) would be more readable

Reply:
Yeah, I can see that. When I started the framework I reference this attribute often, making ideal for a class attribute. But I only reference in one spot now, so it's really not saving me a lot of work. I'll update it.

Reply:
Done